File: libs/java/UtilityLibrary.java
import java.io.File;

public class UtilityLibrary {
    public String concatenateStrings(String... args) {
        StringBuilder sb = new StringBuilder();
        for (String arg : args) {
            sb.append(arg).append(" ");
        }
        return sb.toString().trim();
    }

    public String[] splitString(String input, String delimiter) {
        return input.split(delimiter);
    }

    public String getFileExtension(String filename) {
        File file = new File(filename);
        String extension = "";
        int i = file.getName().lastIndexOf('.');
        if (i > 0) {
            extension = file.getName().substring(i);
        }
        return extension;
    }
}

File: libs/python/UtilityLibrary.py
import os

class UtilityLibrary:
    def concatenate_strings(self, *args):
        """Concatenates multiple strings into one."""
        return ','.join(args)

    def split_string(self, string, delimiter=' '):
        """Splits a string by the given delimiter."""
        return string.split(delimiter)

    def get_file_extension(self, filename):
        """Returns the file extension of the given filename."""
        return os.path.splitext(filename)[1]

